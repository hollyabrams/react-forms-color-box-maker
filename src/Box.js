function Box({ id, width, height, backgroundColor, removeBox }) {
  const boxStyle = {
    width: `${width}px`,
    height: `${height}px`,
    backgroundColor: backgroundColor,
    display: 'inline-block',
    margin: '10px',
  };

  const handleRemove = () => {
    removeBox(id);
  };

  return (
    <div style={boxStyle}>
      <button onClick={handleRemove}>X</button>
    </div>
  );
}

export default Box;
