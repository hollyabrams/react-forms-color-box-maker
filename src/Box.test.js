import { render } from '@testing-library/react';
import Box from './Box';

describe('Box component', () => {
  // Smoke test
  it('renders without crashing', () => {
    const { getByText } = render(<Box id="1" width={100} height={100} backgroundColor="red" />);
    const removeButton = getByText(/X/i);
    expect(removeButton).toBeInTheDocument();
  });

  // Snapshot test
  it('matches snapshot', () => {
    const { container } = render(<Box id="1" width={100} height={100} backgroundColor="red" />);
    expect(container).toMatchSnapshot();
  });
});
