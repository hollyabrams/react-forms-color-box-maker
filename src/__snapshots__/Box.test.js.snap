// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component matches snapshot 1`] = `
<div>
  <div
    style="width: 100px; height: 100px; background-color: red; display: inline-block; margin: 10px;"
  >
    <button>
      X
    </button>
  </div>
</div>
`;
